### ISSUE TITLE
Rename Wasp's TS config to Wasp TS Spec in all places

### ISSUE THREAD

**Issue by sodic** [wasp-lang/wasp#2391]
First, we called it TS SDK, which was confusing (and conflicting with our other SDK, which users import from).

Then we called it TS config, which is confusing because it conflicts with `tsconfig.json`. 

We should come up with a good name for it. We did - it's TS Spec

Update the docs too: https://wasp.sh/docs/general/wasp-ts-config

**Comment by Martinsos** [comment]
- Wasp TS DSL
- Wasp TS Schema -> blah, no
- Wasp TS Definition

We can also try to imagine we have Python alternative in the future, so it should also work for that. So
- Wasp Python DSL
- Wasp Python Definition

It seems to me like DSL is the best but I wish I could think of something nicer.

**Comment by infomiho** [comment]
When I think about it, it's a Wasp Config file in the end. The language seems like a implementation details. 

So it could be:
- Wasp Config in Typescript
- Wasp Config in Python
- Wasp Config in our DSL

Pulumi calls it [project file](https://www.pulumi.com/docs/iac/concepts/projects/project-file/) and [Encore](https://encore.dev/) doesn't really call it anything, just says, there is a `encore.app` file 😂

**Comment by Martinsos** [comment]
Ok there is away to go about it also! However "Wasp Config in Typescript" is a bit clunky, you will want to call it somehow shorter, and there is no way to do that. Also config sounds a bit overly simplistic hm. We will have multiple files soon, modules and what not. I am not saying this might not be the best name though, I am just pointing out cons.

But another idea that I only now remembered -> We have AppSpec, and Wasp itself is Web App SPecification -> so what about "Specification"?

Typescript (Wasp) Spec(ification)
DSL (Wasp) Spec
Python (Wasp) Spec

**Comment by sodic** [comment]
Wasp Spec vs TypeScript Spec sound great!

**Comment by fecony** [comment]
boisdk

**Comment by Martinsos** [comment]
Boris's suggestion: Wasp Typescript app definition

**Comment by sodic** [comment]
@Martinsos I believe we decided on Wasp TS Spec.

**Comment by Martinsos** [comment]
> [@Martinsos](https://github.com/Martinsos) I believe we decided on Wasp TS Spec.

Yes that is it! And we call those spec files (or wasp files). But not config files -> spec files. Wasp at the end is "Web App SPec".

**Comment by developerzeke** [comment]
Should this be closed? (might as well get the issue counter lower)

**Comment by Martinsos** [comment]
I am all in for getting the issue count lower :D! But this one might have some value yet: let me check with @sodic and @FranjoMindek ! I think we haven't yet updated the name everywhere, Franjo and Filip, did we? So should we keep this open till then?

**Comment by sodic** [comment]
Yes, still not updated everywhere. I'll change the issue name, though. Thanks @zkmrtn8!

**Comment by developerzeke** [comment]
It's called TS SDK in some places as well

**Comment by sodic** [comment]
We are going with Wasp spec (Wasp TS spec for now maybe), and the package will be called `@wasp/spec` (coverd in #3782).

**Comment by cprecioso** [comment]
We decided on `@wasp.sh/spec` to be consistent with our current `@wasp.sh/wasp-cli` package. We'll handle the migration to the `@wasp` org in the future (https://github.com/wasp-lang/wasp/issues/4150).

**Comment by cprecioso** [comment]
Done in #4153
